core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -131,58 +132,127 @@ public static <T> void prepareAuthorizationHooks(T manager, DispatcherHooks hook
       hooks.addPostHook(
           "createMetalake",
```

Comment:
hi @jerqi  I think using function name string comparing is too error-prone.

We can create a `SetOwner` annotation to supports, for example:
```
public class MetalakeManager implements MetalakeDispatcher {
  @Override
  @SetOwner
  public BaseMetalake createMetalake(
      NameIdentifier ident, String comment, Map<String, String> properties)
      throws MetalakeAlreadyExistsException {
    long uid = idGenerator.nextId();
   ......
```

Use `SetOwner` annotation to comparing
```
if (method.isAnnotationPresent(SetOwner.class)) { 
NameIdentifier identifier = (NameIdentifier) ((Object[]) args)[0];
                            GravitinoEnv.getInstance()
                                    .ownershipManager()
                                    .setOwner(
                                            identifier,
                                            Entity.EntityType.METALAKE,
                                            PrincipalUtils.getCurrentUserName(),
                                            Owner.Type.USER);
}
```

Reply:
I will implement different dispatchers to refactor this code in the next pull request.